mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -55,86 +56,168 @@ static cl::opt<std::string> outputFilename("o", cl::desc("Output filename"),
                                            cl::value_desc("filename"),
                                            cl::init("-"));
 
-static cl::opt<int> blockSize("block_size", cl::desc("Block size"),
+static cl::opt<std::string> kernelPipeline(
+    "kernel-pipeline", cl::desc("mlir-miopen-driver kernel pipeline list"),
+    cl::value_desc(
+        "comma separated list of miopen pipelines: tuning,gpu,rocdl,binary"),
+    cl::init(""));
+
+static cl::opt<std::string>
+    hostPipeline("host-pipeline",
+                 cl::desc("mlir-miopen-driver host pipeline list"),
+                 cl::value_desc("comma separated list of miopen pipelines: "
+                                "partition,highlevel,execmodel"),
+                 cl::init(""));
+
+static cl::opt<bool> legacyMiopenPipeline("c", cl::Hidden, cl::init(false),
+                                          cl::Optional,
+                                          cl::cb<void, bool>([](bool v) {
+                                            if (v) {
+                                              kernelPipeline.setValue("gpu");
+                                            }
+                                          }));
```

Comment:
It is a confusing way to handle it, anyone who does not review this PR may not know what legacy means. Besides, you are not really using this as a real option but a convenient way to set another option. Since both are static variables, the sequence which one get initialized may be different from your expectation thus creating surprises.

I don't think it is a huge problem if we need to update all tests, however it will be a serious pollution to this PR's diff. Could you do the following:
  - In this PR, revert the `kernelPipeline` to `loweringTargetDialect`. This allows you to just extend rather than break tests.
  - Once this PR is merged, rename the `loweringTargetDialect` to `kernelPipeline`, and update all tests. Do all this and only this change in the new PR such that it is possible to compose/review cleanly.

Reply:
I can do that but I really see option -c as a short-hand or alias anyway, which depends on -target to determine which pipeline gets run.

Reply:
If you want a short hand that's fine, however I think they way this is implemented will cause OB.

```cpp
if (v) {
  kernelPipeline.setValue("gpu");
}
```

Now `kernelPipeline` is also a static variable. The code chunk above assumes one static variable `kernelPipeline` to be initialized before another static variable `legacyMiopenPipeline`. I believe using it this way caused undefined behavior. What would happen if `kernelPipeline` is not initialized if you insist on using it this way?

Reply:
Actually no that is a lambda that is only called if -c is encountered in cl::ParseCommandLineOptions.

Reply:
Well, you may want to run only the host-pipeline or you may want to run specific passes, so defaults to no kernel passes. 

BTW, -c was an alias already and I also use the technique in miopen-gen for short-hand options like -pv.